Sources/Swift/Helper/SentryCurrentDateProvider.swift

```diff
@@ -0,0 +1,20 @@
+@_implementationOnly import _SentryPrivate
+
+@objcMembers class SentryCurrentDateProvider: NSObject {
+
+    func date() -> Date {
+        return Date()
+    }
+
+    func dispatchTimeNow() -> DispatchTime {
+        return .now()
+    }
```

Comment:
l: Are we really not using this? At least we dont touch it during tests.
If so I would remove it.